pkg/reconciler/v1alpha1/revision/reconcile_resources.go

```diff
@@ -147,6 +147,22 @@ func (c *Reconciler) reconcileKPA(ctx context.Context, rev *v1alpha1.Revision) e
 		return fmt.Errorf("Revision: %q does not own PodAutoscaler: %q", rev.Name, kpaName)
 	}
 
+	// Perhaps tha KPA spec changed underneath ourselves?
+	// TODO(vagababov): required for #1997. Should be removed in 0.7,
+	// to fix the protocol type when it's unset.
+	tmpl := resources.MakeKPA(rev)
+	want := kpa.DeepCopy()
+	want.Spec = tmpl.Spec
+	if !equality.Semantic.DeepEqual(want, kpa) {
```

Comment:
Why not just compare `.Spec` before copying?

Reply:
I had it originally so, but it seems we usually do this way.
Changed to check first.